### ISSUE TITLE
Invalid inputs detected for unnamed_Conv_0

### ISSUE THREAD

**Issue by jmswaney** [Microsoft/onnxjs#103]
I'm trying to use an audio CNN from pytorch in onnxjs, but I think I'm having some problems with the Conv1d layers. I got no errors with `onnx.checker.check_model(model)` in python and it seems to load fine into an InferenceSession with `await session.loadModel(...)` in node. However, when I provide the same size and type input as was used for the trace, I get this error:

`Error: invalid inputs detected; op: unnamed_Conv_0 at ExecutionPlan.<anonymous> (C:\Users\...\my_onnx\node_modules\onnxjs\lib\execution-plan.js:166:63)`

The first conv layer in the model corresponds to this layer in ONNX:

`%13 : Float(1, 32, 11024) = onnx::Conv[dilations=[1], group=1, kernel_shape=[7], pads=[0, 0], strides=[4]](%input, %layer_0, %layer_1), scope: ConvEmbedding2/Conv1d[conv1]`

When I remove the conv layers and just use `nn.Linear` and ReLU activations, I have no problems--which suggested to me that I actually made the input tensor correctly but still the conv layers are problematic.

**Comment by fs-eire** [comment]
could you share the input shape and data type of the `Conv`'s inputs X, W and B(optional) ?

**Comment by hariharans29** [comment]
Could you also please share the model file (the one with the conv layer) along with the input type and shape as requested above ?

**Comment by jmswaney** [comment]
Here's part of the graph printed out by onnx:
```
graph torch-jit-export (
  %input[FLOAT, 1x2x44100]
) initializers (
  %layer_0[FLOAT, 32x2x7]
  %layer_1[FLOAT, 32]
  %layer_2[FLOAT, 64x32x31]
  %layer_3[FLOAT, 64]
  %layer_4[FLOAT, 128x64]
  %layer_5[FLOAT, 128]
  %layer_6[FLOAT, 2x128]
  %layer_7[FLOAT, 2]
  %layer_8[FLOAT, 128x2]
  %layer_9[FLOAT, 128]
  %layer_10[FLOAT, 9x128]
  %layer_11[FLOAT, 9]
) {
  %13 = Conv[dilations = [1], group = 1, kernel_shape = [7], pads = [0, 0], strides = [4]](%input, %layer_0, %layer_1)
  %14 = Relu(%13)
  %15 = MaxPool[kernel_shape = [8], pads = [0, 0], strides = [8]](%14)
  %16 = Conv[dilations = [1], group = 1, kernel_shape = [31], pads = [0, 0], strides = [4]](%15, %layer_2, %layer_3)
  %17 = Relu(%16)
...
```
There are no more convs after that, and all of the inputs, weights, and biases seem to be float32's. . Here is the toy model I've been trying to get working:
[model_with_convs.zip](https://github.com/Microsoft/onnxjs/files/2941560/model_with_convs.zip)

To call it in onnx.js, I've been making a `Float32Array` and turning it into a `Tensor` with the same shape as the dummy input used for the trace, which was `[1, 2, 44100]` (one second of stereo audio).

**Comment by jmswaney** [comment]
Update: when I add a channel to make the model take in `[1, 2, 2, 44100]` with `Conv2d`, it's working fine.

**Comment by hariharans29** [comment]
That makes sense as for `Conv2d` the input data `X` has to be necessarily 4D (batch_size, num_channels, height, width). You can deduce that from the spec [here](https://github.com/onnx/onnx/blob/rel-1.2.3/docs/Operators.md#Conv). So adding a dim with unit value to the Tensor (for the same data) makes it cross the input validation step (which was the original error). 

However, please check the correctness of the result because it can also be `[1, 2, 44100, 1]` depending on how the model has been trained.

Closing this issue for now. Please re-open if needed.
